clang/include/clang/Basic/DiagnosticGroups.td

```diff
@@ -1252,3 +1252,7 @@ in addition with the pragmas or -fmax-tokens flag to get any warnings.
 def WebAssemblyExceptionSpec : DiagGroup<"wasm-exception-spec">;
 
 def RTTI : DiagGroup<"rtti">;
+
+// A warning group for warnings related to constexprs captured as
+// kernel args.
+def ConstExprKernelArg : DiagGroup<"constexpr-kernel-arg">;
```

Comment:
Were you planning to add more diagnostics to this group?  If not, I think we could defer having a group for this until later.

Reply:
I didn't have a new group added initially but then "warning-flags.c" failed and looking at that file's description, it suggested that I add anew group for warnings without a dedicated flag.
See: [L1516](https://github.com/intel/llvm/blob/sycl/clang/test/Misc/warning-flags.c#L15)

Reply:
I can't think of an existing flag to reuse at the moment, so this is okay.  But it would be cleaner to avoid it.

Reply:
I feel adding this group adds at least -W flag for the new diagnostic. "warning-flags.c" test says that we shouldn't add the new warnings without flags, i.e. we probably need some group anyway. So, I think we need to understand what do we want to do. For example we have some -sycl-strict diagnostic group, but warnings from this group are disabled by default by the Driver, if we are fine with this behavior, we can re-use sycl-strict group. If we want our warning to be enabled by default, we could add a special group for it. Or, for example, this warning says about possible non-obvious performance degradation to user, I think we may have another cases of similar non-obvious performance degradations as well as another warnings for such cases, so we can allocate the group for them.

Reply:
I feel this warning should be enabled by default, as like you say, it detects something immediately non-obvious and likely helpful for performance reasons. Adding a group for this purpose would be great.   Does the name have to be decided now?  Will choosing the group name at a later stage be considered "breaking" as the old spelling will no longer work?

Reply:
> Does the name have to be decided now? Will choosing the group name at a later stage be considered "breaking" as the old spelling will no longer work?

I think it is better to choose the name now, so we don't break user's code in the future.

Reply:
Any suggestions for the generic group name that would capture all non-obvious warnings? <nonobvious-perf-degradation>?

Reply:
I'm not really good at this but, how about something like `perf-impact`?

Reply:
Works for me, unless @premanandrao or @elizabethandrews have other suggestions. If I don't hear any other options by tomorrow, I will update the group name to `perf-impact`.
Thanks, @Fznamznon

Reply:
I am okay with that spelling.